lib/listings/useFetchCurrentUserListing.tsx

```diff
@@ -0,0 +1,12 @@
+import { useQuery } from "react-query";
+import ListingService from "./ListingService";
+import {
+    IBaseListing,
+    IListingAttributes,
+    TListingCollectionReference,
+    TListingDocumentReference,
+  } from "./types"
+
+export default function useFetchCurrentUserListing(uid: any) {
+  return useQuery(["listings"], () => ListingService.fetchcurrentuserListings(uid));
```

Comment:
we should add the userId in this query key, since we are querying a subcollection

Reply:
done